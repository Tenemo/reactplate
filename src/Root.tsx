import React, { ReactElement } from 'react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { HelmetProvider } from 'react-helmet-async';

import { store, history } from 'store/configureStore';
import App from 'components/App';

const Root: React.FunctionComponent = (): ReactElement => (
    <Provider store={store}>
        <ConnectedRouter history={history}>
            <HelmetProvider>
                <App />
            </HelmetProvider>
        </ConnectedRouter>
    </Provider>
);

export default Root;
