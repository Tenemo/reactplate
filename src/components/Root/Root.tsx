import React, { ReactElement, useEffect } from 'react';
import { Provider } from 'react-redux';
import { HistoryRouter as Router } from 'redux-first-history/rr6';
import { HelmetProvider } from 'react-helmet-async';

import { store, history } from 'store/configureStore';

import App from 'components/App';
import 'styles/global.scss';

export const Root = (): ReactElement => {
    useEffect(() => {
        // https://stackoverflow.com/questions/31402576/enable-focus-only-on-keyboard-use-or-tab-press
        document.body.addEventListener('mousedown', () =>
            document.body.classList.add('using-mouse'),
        );
        document.body.addEventListener('keydown', (event) => {
            if (event.key === 'Tab') {
                document.body.classList.remove('using-mouse');
            }
        });
    }, []);

    return (
        <Provider store={store}>
            <HelmetProvider>
                <Router history={history}>
                    <App />
                </Router>
            </HelmetProvider>
        </Provider>
    );
};

export default Root;
