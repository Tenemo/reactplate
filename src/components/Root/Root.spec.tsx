import React from 'react';
import { shallow } from 'enzyme';

import { Root } from './Root';

describe('Root', () => {
    it('matches snapshot', () => {
        const wrapper = shallow(<Root />);
        expect(wrapper).toMatchSnapshot();
    });
    it('renders Redux Provider', () => {
        const wrapper = shallow(<Root />);
        const Provider = wrapper.find('Provider');
        expect(Provider.exists()).toEqual(true);
    });
    it('renders ConnectedRouter', () => {
        const wrapper = shallow(<Root />);
        const ConnectedRouter = wrapper.find(
            'Connect(ConnectedRouterWithContext)',
        );
        expect(ConnectedRouter.exists()).toEqual(true);
    });
    it('renders App', () => {
        const wrapper = shallow(<Root />);
        const App = wrapper.find('App');
        expect(App.exists()).toEqual(true);
    });
});
