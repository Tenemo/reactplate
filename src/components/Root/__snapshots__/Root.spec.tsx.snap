// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root matches snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <HelmetProvider
    context={Object {}}
  >
    <Connect(ConnectedRouterWithContext)
      history={
        Object {
          "action": "POP",
          "back": [Function],
          "block": [Function],
          "createHref": [Function],
          "forward": [Function],
          "go": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "default",
            "pathname": "/",
            "search": "",
            "state": null,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <App />
    </Connect(ConnectedRouterWithContext)>
  </HelmetProvider>
</Provider>
`;
